### PR TITLE
Add zarpaint recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~[ ] If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

---

Depends on:
- https://github.com/conda-forge/staged-recipes/pull/17848

### CODE REVIEW

**Comment by goanpeca** [comment]
Depends on:
- https://github.com/conda-forge/staged-recipes/pull/17848

**Comment by goanpeca** [comment]
We are in the process of creating conda packages for [napari plugins](https://www.napari-hub.org/) and related dependencies. Once this recipe passes all the checks, every time a new release goes into PyPI, a PR will be automatically open, with usually very minor need for changes so the process of maintaining this moving forward is small. Also the @conda-forge/napari team will be available to help as needed.

You can read a bit more about conda-forge at https://conda-forge.org/#contribute

--- 

@jni, do you agree with being added as a maintainer?

**Comment by jni** [comment]
Yes.

btw @goanpeca I may have accidentally buried some of these that belong to me because I got notified for all of them. Please ping me on Zulip if I haven't responded on one of my plugins.

**Comment by goanpeca** [comment]
Will do!

**Comment by jni** [comment]
@goanpeca @jaimergp this depends on tensorstore right now (#17848), though we're in the process of removing that dependency... Still, it would be good to get that in as it's a very good library. 🙏

**Comment by goanpeca** [comment]
Still working on this!

**Comment by jni** [comment]
> Still working on this!

oh good! 😂

**Comment by jni** [comment]
@goanpeca I'm working on a new zarpaint release for which tensorstore is optional. I'll update this recipe once that's out. 🙏

File: recipes/zarpaint/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{% set name = "zarpaint" %}
+{% set version = "0.2.0" %}
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:2]:
```suggestion
{% set version = "0.3.0" %}
```

File: recipes/zarpaint/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/zarpaint-{{ version }}.tar.gz
+  sha256: f73f80d9d517f81a89a5b5b1b15f753ee436e9b1fb8bb404e01c1fc1e67ba714
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:10]:
```suggestion
  sha256: c831eabfe5148d1a41ef29fe2a8fc7da7e61b4bdf8a910b73aaad70730169f03
```

File: recipes/zarpaint/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+    - setuptools-scm
+  run:
+    - magicgui
+    - napari >=0.4.11.dev80
+    - napari-plugin-engine >=0.1.4
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:27]:
```suggestion
```

File: recipes/zarpaint/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+    - setuptools-scm
+  run:
+    - magicgui
+    - napari >=0.4.11.dev80
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:26]:
```suggestion
    - napari >=0.4.18
```

File: recipes/zarpaint/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    - napari-plugin-engine >=0.1.4
+    - numpy
+    - python >=3.7
+    - pyyaml
+    - qtpy
+    - scikit-image
+    - scipy
+    - tensorstore
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:34]:
```suggestion
```

File: recipes/zarpaint/meta.yaml (hunk 3 above)
Comment (jni) [reply recipes/zarpaint/meta.yaml:27]: (I just saw that the setup.cfg has it but it's a mistake)

**Review by jni — CHANGES_REQUESTED** [review]
@goanpeca new release of zarpaint is out, drops the tensorstore requirement! 🥳

**Comment by goanpeca** [comment]
```bash
+ zarpaint --help
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/zarpaint_1699311916585/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/bin/zarpaint", line 7, in <module>
    from zarpaint._main import main
  File "/home/conda/staged-recipes/build_artifacts/zarpaint_1699311916585/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.11/site-packages/zarpaint/_main.py", line 5, in <module>
    from ._zarpaint import correct_labels
ImportError: cannot import name 'correct_labels' from 'zarpaint._zarpaint' (/home/conda/staged-recipes/build_artifacts/zarpaint_1699311916585/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.11/site-packages/zarpaint/_zarpaint.py)

```

@jni, is this expected 🙃 ?

**Comment by jni** [comment]
Hahahaha no it is not! Hold please!

File: recipes/zarpaint/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    - toolz
+    - zarr
+
+test:
+  imports:
+    - zarpaint
+  commands:
+    - pip list
+    - pip check
+    - zarpaint --help
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:42]:
```suggestion
```

File: recipes/zarpaint/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,58 @@
+{% set name = "zarpaint" %}
+{% set version = "0.3.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/zarpaint-{{ version }}.tar.gz
+  sha256: c831eabfe5148d1a41ef29fe2a8fc7da7e61b4bdf8a910b73aaad70730169f03
```
Comment (jni) [inline recipes/zarpaint/meta.yaml:10]:
```suggestion
{% set version = "0.3.1" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/zarpaint-{{ version }}.tar.gz
  sha256: dda1352084afad3ea6f0bbfa25c5cec0e8077ccdd34022a12a32bd661af0d41c
```

**Review by jni — APPROVED** [review]
Let's try now... 😅

**Comment by jni** [comment]
To paraphrase Gordon Gecko, green is good! 😃

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

**Pull request merged**